packages/compose/src/hooks/use-focus-outside/test/index.js

```diff
@@ -4,120 +4,149 @@
-		input.focus();
-		input.blur();
+		// Click and focus button inside the wrapper
+		await user.click(
+			screen.getByRole( 'button', { name: 'Button inside the wrapper' } )
+		);
 
-		jest.runAllTimers();
+		expect( mockOnFocusOutside ).not.toHaveBeenCalled();
 
-		expect( onFocusOutside ).toHaveBeenCalled();
-	} );
+		// Click and focus button outside the wrapper
+		await user.click(
+			screen.getByRole( 'button', { name: 'Button outside the wrapper' } )
+		);
 
-	it( 'should not call handler if focus shifts outside the component when the document does not have focus', () => {
-		render( <TestComponent onFocusOutside={ onFocusOutside } /> );
+		expect( mockOnFocusOutside ).toHaveBeenCalled();
+	} );
 
+	it( 'should not call handler if focus shifts outside the component when the document does not have focus', async () => {
 		// Force document.hasFocus() to return false to simulate the window/document losing focus
 		// See https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus.
-		document.hasFocus = () => false;
+		const mockedDocumentHasFocus = jest
+			.spyOn( document, 'hasFocus' )
+			.mockImplementation( () => false );
+		const mockOnFocusOutside = jest.fn();
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
 
-		const input = screen.getByRole( 'textbox' );
-		input.focus();
-		input.blur();
+		render(
+			<FocusOutsideComponent onFocusOutside={ mockOnFocusOutside } />
+		);
 
-		jest.runAllTimers();
+		// Click and focus button inside the wrapper, then click and focus
+		// a button outside the wrapper.
+		await user.click(
+			screen.getByRole( 'button', { name: 'Button inside the wrapper' } )
+		);
+		await user.click(
+			screen.getByRole( 'button', { name: 'Button outside the wrapper' } )
+		);
+
+		// The handler is not called thanks to the mocked return value of
+		// `document.hasFocus()`
+		expect( mockOnFocusOutside ).not.toHaveBeenCalled();
 
-		expect( onFocusOutside ).not.toHaveBeenCalled();
+		// Restore the `document.hasFocus()` function to its original implementation.
+		mockedDocumentHasFocus.mockRestore();
 	} );
 
-	it( 'should cancel check when unmounting while queued', () => {
+	it( 'should cancel check when unmounting while queued', async () => {
+		const mockOnFocusOutside = jest.fn();
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
 		const { rerender } = render(
-			<TestComponent onFocusOutside={ onFocusOutside } />
+			<FocusOutsideComponent onFocusOutside={ mockOnFocusOutside } />
 		);
 
-		const input = screen.getByRole( 'textbox' );
-		input.focus();
-		input.blur();
+		// Click and focus button inside the wrapper.
+		const button = screen.getByRole( 'button', {
+			name: 'Button inside the wrapper',
+		} );
+		await user.click( button );
 
+		// Simulate a blur event and the wrapper unmounting while the blur event
+		//  handler is queued
+		button.blur();
 		rerender( <div /> );
```

Comment:
I know this is pre-existing, but I've found [`unmount`](https://testing-library.com/docs/react-testing-library/api/#unmount) to be a more explicit way to express what we're doing here.

Reply:
Interesting! I'll look into it and report back :)

Reply:
Done in https://github.com/WordPress/gutenberg/pull/45317/commits/6309e63206c8b1defcc258e13f316a91c0a75a8f